File: product-catalog/src/main/java/com/ikibm/catalog/constants/CampaignIds.java
package com.ikibm.catalog.constants;

import java.util.List;

/** Vitrin kampanya slider'ındaki sabit kampanya id'leri (seed/campaigns.json ile eşleşir). */
public final class CampaignIds {

    private CampaignIds() {}

    public static final List<String> ALL = List.of(
            "ups-guardian",
            "hik-security",
            "tenda-network",
            "fiber-infra",
            "huawei-enterprise"
    );

    public static boolean isValid(String id) {
        return ALL.contains(id);
    }
}

File: server/src/constants/campaignIds.js
// Ana sayfa kampanya slider'ındaki (CampaignSlider) sabit kampanya id'leri.
// Frontend'deki src/data/campaigns.js ile eşleşir — kampanya görseli yüklenirken
// keyfi bir campaignId ile S3'e yazılmasını engellemek için allowlist görevi görür.
// campaigns.js'e yeni bir kampanya eklendiğinde buraya da eklenmelidir.
export const CAMPAIGN_IDS = [
  "ups-guardian",
  "hik-security",
  "tenda-network",
  "fiber-infra",
  "huawei-enterprise",
];
